File: Recursion/gcd.c
#include <stdio.h>

long int gcd(int a , int b){
    if (b==0){
        return a;
    }
    else{
        return gcd(b , a%b);
    }
}

int main(){
    int a , b , GCD;
    printf("Enter the value of a and b \n");
    scanf("%d , %d" , &a , &b);
    GCD = gcd(a,b);
    printf("The hcf of two numberr is  %d" ,GCD );
    return 0;
}

File: DSA_Lab/Recursion/gcd.py
def GCD(a,b):
    if b==0:
        return a
    else:
        return GCD(b , a%b)
    
def main():
    a , b = int(input("Enter the two numbers a =  ")) , int(input("\nEnter the next number b = "))
    hcf = GCD(a,b)
    print(f"The hcf of {a} , {b} is {hcf}\n")

main()